iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/string.inl

```diff
@@ -321,7 +321,11 @@ inline int64_t string<Capacity>::compare(char other) const noexcept
     auto result = memcmp(c_str(), &other, 1U);
     if (result == 0)
     {
-        if (m_rawstringSize > 1U)
+        if (empty())
+        {
+            return -1;
+        }
+        else if (m_rawstringSize > 1U)
```

Comment:
I would recommend avoiding this kind of `else` branching if there is an early return (which should be used as often as possible for easy to follow code flow and optimization).

No need for changes though and the compiler will optimize the jumps/returns itself in this case.